deployment/vm/vm.go

```diff
@@ -151,6 +179,8 @@ func (vm *vm) AttachDisk(disk bidisk.Disk) error {
 		return bosherr.WrapError(err, "Attaching disk in the cloud")
 	}
 
+	vm.cloud.SetDiskMetadata(disk.CID(), vm.createDiskMetadata())
```

Comment:
i take it we are ignoring error in this case to make this backwards compatible with older CPIs. let's log the error using logger.